monitor/subscription.go

```diff
@@ -137,13 +150,13 @@ func (m *NodeMonitor) Subscribe(ctx context.Context, params *opcua.SubscriptionP
 // via the monitor's `ErrHandler`.
 // The caller must call `Unsubscribe` to stop and clean up resources. Canceling the context
 // will also cause the subscription to stop, but `Unsubscribe` must still be called.
-func (m *NodeMonitor) ChanSubscribe(ctx context.Context, params *opcua.SubscriptionParameters, ch chan<- *DataChangeMessage, nodes ...string) (*Subscription, error) {
-	sub, err := newSubscription(ctx, m, params, 16, nodes...)
+func (m *NodeMonitor) ChanSubscribe(ctx context.Context, params *opcua.SubscriptionParameters, ch chan<- Message, eventSub bool, filter *ua.ExtensionObject, nodes ...string) (*Subscription, error) {
```

Comment:
I am starting to think that an `args` parameter might be useful here. We did something similar with the `WithContext` versions of all functions where we dropped the `WithContext` suffix in a later version. So maybe we can refactor this change so that we add `ChanSubscribeWithArgs(ctx context.Context, args ChanSubscribeArgs) (*Subscription, error)` and drop the `WithArgs` in 0.8.0. Add a deprecation note to the other version. What do you think?